### PR TITLE
Refactoring in typescript

### PR DESCRIPTION
**Pull request by diazod**
- All in typescript now
- Update signature of functions
- Using jest and improvement in tests

### CODE REVIEW

File: src/WhereBuilder.ts [hunk 1]
```diff
@@ -0,0 +1,316 @@
...
+        this.adWhereActionToQueryBuilder(
+          queryBuilder,
+          (ctx: Knex.QueryInterface) => ctx.whereRaw(query, values || []),
+          isFirstClause2
+        );
+      }
+    );
+  }
+
+  private adWhereActionToQueryBuilder(
```
Comment (itereshchenkov) [inline src/WhereBuilder.ts:146]:
```suggestion
  private addWhereActionToQueryBuilder(
```

File: src/Finder.ts [hunk 2]
```diff
@@ -0,0 +1,292 @@
...
+  public getClient() {
+    return this.dbClient;
+  }
+
+  public getPrimaryKey(): string | null {
+    const current = (this.constructor as any) as Finder<T>;
+    return current.primaryKey || this.primaryKey;
+  }
+
+  public getTableName(): string | null {
```
[the remark this answers is not in the document]
Comment (diazod) [reply src/Finder.ts:64]: if it is not define yes

[the remark this answers is not in the document]
Comment (diazod) [reply src/Finder.ts:64]: @itereshchenkov Actually it does crash if there's not configured table name configured:
https://github.com/collectai/base-dao/pull/18/files#diff-8a340b403adac7265842be3c42dc0f43R38

[the remark this answers is not in the document]
Comment (itereshchenkov) [reply src/Finder.ts:64]: I think it makes no sense to use it without defined table name

**Comment by diazod** [comment]
@itereshchenkov @zukomgwili Feedback applied

**Comment by itereshchenkov** [comment]
LGTM

**Pull request merged**